### PR TITLE
TINY-4603: Renamed variables.

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket: 

Description of Changes:
Variable names have changed. Not a new product feature.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

**Review by vpyshnenko — APPROVED** [review]

File: _data/nav.yml [hunk 1]
```diff
@@ -318,7 +318,7 @@
     - url: "anchor"
     - url: "autolink"
       pages:
-      - url: "#default_link_target"
+      - url: "#link_default_target"
```
Comment (lnewson) [inline _data/nav.yml:321]: These need reordering as previously it was in alphabetical order.

- [ ] Reorder the sections so they remain in alphabetical order please

**Review by lnewson — CHANGES_REQUESTED** [review]
This is targeting the wrong branch (it should be targeting the 6.0 feature branch and it also needs some migration/release notes as well.

**Comment by HAFRMO** [comment]
The addition to the migration-page was left purposefully simple, as was requested in the ticket.

**Comment by tylerkelly13** [comment]
@HAFRMO looking good so far, still needs release notes.

**Review by lnewson — CHANGES_REQUESTED** [review]
As Tyler mentioned part of the work needed is still missing.

**Comment by tylerkelly13** [comment]
Also, this branch has a build failure:
```
Could not locate the included file 'configuration/link_default_target.md' in any of ["/pipeline/source/_includes"]. Ensure it exists in one of those directories and, if it is a symlink, does not point outside your site source. in plugins-ref/opensource/autolink.md
```

File: release-information/release-notes/6_0_0/6_0_0-release-notes-core-changes.md [hunk 2]
```diff
@@ -39,3 +39,9 @@ The following enhancements were made for the {{site.productname}} 6.0 release.
 The following functionality changes were made for the {{site.productname}} 6.0 release:
 
 - changelog
+
+### Renamed Options
+
+- The `default_link_target` option has been renamed to `link_default_target` for both `link` and `autolink` plugins
+- The `rel_list` option has been renamed to `link_rel_list` for the `link` plugin
+- The `target_list` option has been renamed to `link_target_list` for the `link` plugin
```
Comment (lnewson) [inline release-information/release-notes/6_0_0/6_0_0-release-notes-core-changes.md:47]:
```suggestion
- The `default_link_target` option has been renamed to `link_default_target` for both `link` and `autolink` plugins.
- The `rel_list` option has been renamed to `link_rel_list` for the `link` plugin.
- The `target_list` option has been renamed to `link_target_list` for the `link` plugin.
```

File: how-to-guides/migration-from-5x.md [hunk 3]
```diff
@@ -31,3 +31,9 @@ To use TinyMCE 6 from the {{site.cloudname}}, include this script in your HTML p
 ```
 
 For information on using the latest development and testing builds, see: [Cloud deployment guide - Specify editor & plugin versions]({{site.baseurl}}/cloud-deployment-guide/editor-plugin-version/).
+
+### Renamed Options
+
+* The `default_link_target` option has been renamed to `link_default_target` for both `link` and `autolink` plugins
+* The `rel_list` option has been renamed to `link_rel_list` for the `link` plugin
+* The `target_list` option has been renamed to `link_target_list` for the `link` plugin
```
Comment (lnewson) [inline how-to-guides/migration-from-5x.md:39]:
```suggestion
* The `default_link_target` option has been renamed to `link_default_target` for both `link` and `autolink` plugins.
* The `rel_list` option has been renamed to `link_rel_list` for the `link` plugin.
* The `target_list` option has been renamed to `link_target_list` for the `link` plugin.
```

**Review by lnewson — APPROVED** [review]

**Review by tylerkelly13 — APPROVED** [review]

**Review by EkimChau — APPROVED** [review]

**Pull request merged**